sdjournal/journal_test.go

```diff
@@ -343,3 +344,54 @@ func TestJournalReaderSmallReadBuffer(t *testing.T) {
 		t.Fatalf("Got unexpected message %s", got[1])
 	}
 }
+
+func TestJournalGetUniqueValues(t *testing.T) {
+	j, err := NewJournal()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer j.Close()
+
+	uniqueString := generateRandomField(20)
+	testEntries := []string{"A", "B", "C", "D"}
+	for _, v := range testEntries {
+		err := journal.Send("TEST: "+uniqueString, journal.PriInfo, map[string]string{uniqueString: v})
+		if err != nil {
+			t.Fatal(err)
+		}
+		time.Sleep(time.Millisecond * 250)
```

Comment:
> unfortunately waitAndNext is racy when we send several messages in a loop. Sometimes the test passes and sometime does not.

This is bad news, the wait is probably getting awaken by another event before your entries are hitting the journal.

I see you are basically implementing a busy loop around `GetUniqueValues()`, which is still a crude hack around missing a proper wait message. I don't want to cause more pain on this PR, so let's just go back to a plain sleep for now and we'll follow-up with a proper `waitOnMatch()` here. Can you please revert to sleeping like in the initial iteration of this PR, and leave there a TODO for the missing helper? And sorry again for the back and forth.